src/api/endpoint/api/v1/search.ts

```diff
@@ -0,0 +1,96 @@
+import semver from 'semver'
+
+function compileTextSearch(textSearch) {
+        const personMatch = (person, search) => {
+            if(typeof person === 'string')
+                return person.includes(search);
+            
+            if(typeof person === 'object')
+                for(const field of Object.values(person))
+                    if(typeof field === 'string' && field.includes(search))
+                        return true;
+            
+            return false;
+        }
+        const matcher = function(q) {
+            const match = q.match(/author:(.*)/)
+            if(match !== null)
+                return (pkg) => personMatch(pkg.author, match[1])
+
+            // TODO: maintainer, keywords, not/is unstable insecure, boost-exact
+            // TODO implement some scoring system for freetext
+            return (pkg) => {
+                return ['name', 'displayName', 'description']
+                    .map(k => pkg[k])
+                    .filter(x => x !== undefined)
+                    .some(txt => txt.includes(q))
+            };
+        }
+
+        const textMatchers = (textSearch || '').split(' ').map(matcher);
+        return (pkg) => textMatchers.every(m => m(pkg));
+}
+
+export default function(route, auth, storage): void {
+    route.get('/-/v1/search', (req, res)=>{
+        let [text, size, from, quality, popularity, maintenance] = 
```

Comment:
Sonarcloupd reports that `quality, popularity, maintenance` are not used. We should remove this so we can make him happy again.

Reply:
> Sonarcloupd reports that `quality, popularity, maintenance` are not used. We should remove this so we can make him happy again.

I think this depends on if you want to implement the result scoring part of this endpoint or stay with the stubbed static scores.

Reply:
If is too much work, I would ad a comment Todo and do it later. I think basic functionality works for now. IMHO

Reply:
Agreed, as I am using verdaccio exclusively for hosting private repositories for Untiy3D packages, I personally have no use for the scoring feature